src/drivers/imx/esai.c

```diff
@@ -194,41 +194,48 @@ static inline int esai_set_config(struct dai *dai,
 
 	xccr |= ESAI_xCCR_xHCKD; /* Set the HCKT pin as an output */
 
-	dai_update_bits(dai, REG_ESAI_TCCR, mask, xccr); /* rx */
-	dai_update_bits(dai, REG_ESAI_RCCR, mask, xccr); /* tx */
+	dai_update_bits(dai, REG_ESAI_TCCR, mask, xccr);
+	/* There is a hardware limitation which prevents tx and rx to be
+	 * simultaneously master or simultaneously slave. As a workaround,
+	 * we will leave tx as master and set rx as slave.
+	 */
+	xccr &= ~(ESAI_xCCR_xCKD | ESAI_xCCR_xFSD);
+	dai_update_bits(dai, REG_ESAI_RCCR, mask, xccr);
 
 	mask = ESAI_xCR_xFSL | ESAI_xCR_xFSR | ESAI_xCR_xWA |
 		ESAI_xCR_xMOD_MASK | ESAI_xCR_xSWS_MASK | ESAI_xCR_PADC |
 		ESAI_xCR_xPR;
 	/* Personal reset, suspend the actual TX/RX for now */
 	xcr |= ESAI_xCR_xPR;
 
-	dai_update_bits(dai, REG_ESAI_TCR, mask, xcr); /* rx */
-	dai_update_bits(dai, REG_ESAI_RCR, mask, xcr); /* tx */
+	dai_update_bits(dai, REG_ESAI_TCR, mask, xcr);
+	dai_update_bits(dai, REG_ESAI_RCR, mask, xcr);
 
```

Comment:
Why do you remove the comments tx, rx here? Please if you want to do cleanups make a separate patch.

Reply:
The comments were wrong as far as I understand right now, so I thought of removing them because they were misleading.